### PR TITLE
Issue 173

### PR DESCRIPTION
**Pull request by icui**
# Description

Basic configuration for pip install

# Issue Number

#173 

# Checklist

Please make sure to check developer documentation on specfem docs. 

[x] I ran the code through pre-commit to check style
[x] My code passes all the integration tests
[x] I have added sufficient unittests to test my changes
[x] I have added/updated documentation for the changes I am proposing
[x] I have updated CMakeLists to ensure my code builds
[x] My code builds across all platforms

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -1,8 +1,43 @@
...
 description = "Kokkos implementation of SpecFEM2D code"
-authors = ["Your Name <you@example.com>"]
+readme = "README.md"
+authors = [
+  { name = "Rohit R Kakodkar", email = "rohit_kakodkar@princeton.edu" },
+  { name = "Lucas Sawade", email = "lsawade@princeton.edu" },
+  { name = "Congyue Cui", email = "ccui@princeton.edu" }
+]
+requires-python = ">=3.9"
+# classifiers = [
```
Comment (Rohit-Kakodkar) [inline pyproject.toml:16]: We can remove these comments

File: pyproject.toml [hunk 2]
```diff
@@ -1,8 +1,43 @@
...
+#   "Programming Language :: Python :: 3.12",
+#   "Programming Language :: Python :: 3.13",
+# ]
+
+[tool.scikit-build]
+cmake.define.CMAKE_BUILD_TYPE = "Release"
+cmake.define.BUILD_TESTS = "ON"
+cmake.define.ENABLE_SIMD = "ON"
+cmake.define.ENABLE_ATOMICS_BYPASS = "ON"
+# cmake.define.Kokkos_ENABLE_OPENMP = "ON"
```
Comment (Rohit-Kakodkar) [inline pyproject.toml:34]: Here as well.

**Review by Rohit-Kakodkar — COMMENTED** [review]
LGTM to merge once the tests pass. Eventually we'd need to create a github action to test the pip install

File: pyproject.toml (hunk 1 above)
Comment (icui) [reply pyproject.toml:16]: The comments will be removed in the next pull request

File: pyproject.toml [hunk 3]
```diff
@@ -1,8 +1,32 @@
...
+]
+requires-python = ">=3.9"
+
+[tool.scikit-build]
+cmake.define.CMAKE_BUILD_TYPE = "Release"
+cmake.define.BUILD_TESTS = "OFF"
+cmake.define.ENABLE_SIMD = "ON"
+cmake.define.ENABLE_ATOMICS_BYPASS = "ON"
+cmake.define.Kokkos_ENABLE_OPENMP = "OFF"
+cmake.define.Kokkos_ARCH_NATIVE = "ON"
```
Comment (Rohit-Kakodkar) [inline pyproject.toml:23]: For distro, we want to build for the lowest possible architecture. So I think the best build recipe is `-D CMAKE_BUILD_TYPE=Release` `-D BUILD_TESTS=ON`. Setting the build tests option on would let use allow users to check their installation

**Comment by Rohit-Kakodkar** [comment]
retest this please

**Pull request merged**